core/src/avm1/globals/array.rs

```diff
@@ -624,10 +610,9 @@ pub fn create_proto<'gc>(
     proto: Object<'gc>,
     fn_proto: Object<'gc>,
 ) -> Object<'gc> {
-    let array = ScriptObject::array(gc_context, Some(proto));
-    let object = array.as_script_object().unwrap();
+    let object = ScriptObject::object(gc_context, Some(proto));
```

Comment:
`proto` has to be an `ArrayObject` for instances to be instantiated correctly (e.g., when doing `new Array("a", "b");`.